Source/WebKitLegacy/mac/History/HistoryPropertyList.mm

```diff
@@ -32,36 +32,38 @@
 
 static const int currentFileVersion = 1;
 
+struct CFMalloc {
+    static void* malloc(size_t size) { return CFAllocatorAllocate(nullptr, size, 0); }
+    static void free(void* p) { CFAllocatorDeallocate(nullptr, p); }
+};
+
 HistoryPropertyListWriter::HistoryPropertyListWriter()
     : m_displayTitleKey("displayTitle"_s)
     , m_lastVisitWasFailureKey("lastVisitWasFailure"_s)
     , m_lastVisitedDateKey("lastVisitedDate"_s)
     , m_redirectURLsKey("redirectURLs"_s)
     , m_titleKey("title"_s)
     , m_urlKey(emptyString())
-    , m_buffer(0)
 {
 }
 
-UInt8* HistoryPropertyListWriter::buffer(size_t size)
+HistoryPropertyListWriter::~HistoryPropertyListWriter() = default;
+
+std::span<UInt8> HistoryPropertyListWriter::buffer(size_t size)
 {
     ASSERT(!m_buffer);
-    m_buffer = static_cast<UInt8*>(CFAllocatorAllocate(0, size, 0));
-    m_bufferSize = size;
-    return m_buffer;
+    m_buffer = MallocSpan<UInt8, CFMalloc>::malloc(size);
+    return m_buffer.mutableSpan();
 }
 
 RetainPtr<CFDataRef> HistoryPropertyListWriter::releaseData()
 {
-    UInt8* buffer = m_buffer;
-    if (!buffer)
-        return 0;
-    m_buffer = 0;
-    RetainPtr<CFDataRef> data = adoptCF(CFDataCreateWithBytesNoCopy(0, buffer, m_bufferSize, 0));
-    if (!data) {
-        CFAllocatorDeallocate(0, buffer);
-        return 0;
-    }
+    if (!m_buffer)
+        return nullptr;
+    auto span = m_buffer.leakSpan();
+    RetainPtr data = adoptCF(CFDataCreateWithBytesNoCopy(0, span.data(), span.size(), 0));
```

Comment:
This is not new to your patch, but how does the CFData end up freeing the span it adopts? I would imagine that BytesNoCopy prevents calling the deallocator.

Reply:
No, the deallocator is called even if we use this create function. If we didn't want it to deallocate we'd have to pass `kCFAllocatorNull`. Here we pass `kCFAllocatorDefault`, although we write it `0`.

Reply:
A special thing about the `NoCopy` function is that caller passes a separate deallocator for the bytes, the last argument to the function, whereas the other functions use the same allocator that is used to allocate the structure itself, the first argument to the function, for both allocating and deallocating the bytes.